libraries/apollo-runtime-java/api/apollo-runtime-java.api

```diff
@@ -9,22 +9,31 @@ public abstract interface class com/apollographql/apollo3/runtime/java/ApolloCal
 
 public class com/apollographql/apollo3/runtime/java/ApolloClient {
 	public fun execute (Lcom/apollographql/apollo3/api/ApolloRequest;Lcom/apollographql/apollo3/runtime/java/ApolloCallback;)Lcom/apollographql/apollo3/runtime/java/interceptor/ApolloDisposable;
+	public fun getCustomScalarAdapters ()Lcom/apollographql/apollo3/api/CustomScalarAdapters;
 	public fun mutation (Lcom/apollographql/apollo3/api/Mutation;)Lcom/apollographql/apollo3/runtime/java/ApolloCall;
 	public fun query (Lcom/apollographql/apollo3/api/Query;)Lcom/apollographql/apollo3/runtime/java/ApolloCall;
+	public fun subscription (Lcom/apollographql/apollo3/api/Subscription;)Lcom/apollographql/apollo3/runtime/java/ApolloCall;
 }
 
 public class com/apollographql/apollo3/runtime/java/ApolloClient$Builder {
 	public fun <init> ()V
 	public fun addCustomScalarAdapter (Lcom/apollographql/apollo3/api/CustomScalarType;Lcom/apollographql/apollo3/api/Adapter;)Lcom/apollographql/apollo3/runtime/java/ApolloClient$Builder;
 	public fun addInterceptor (Lcom/apollographql/apollo3/runtime/java/interceptor/ApolloInterceptor;)Lcom/apollographql/apollo3/runtime/java/ApolloClient$Builder;
 	public fun addInterceptors (Ljava/util/List;)Lcom/apollographql/apollo3/runtime/java/ApolloClient$Builder;
+	public fun addWsHeader (Lcom/apollographql/apollo3/api/http/HttpHeader;)Lcom/apollographql/apollo3/runtime/java/ApolloClient$Builder;
+	public fun addWsHeaders (Ljava/util/List;)Lcom/apollographql/apollo3/runtime/java/ApolloClient$Builder;
```

Comment:
Would it make send to duplicate the Kotlin API here? (with NetworkTransport as an interface, public WebSocketNetworkTransport etc..?). Less cognitive load for us and for the users?

Reply:
I think probably yes :) Will amend.

Reply:
Turns out we have one already: https://github.com/apollographql/apollo-kotlin/blob/05f2a3295517fb7a64c2259c38b453a37f3c4d19/libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/ImmutableMapBuilder.kt#L6. (I thought it was a JDK thing but it's not, it's coming [from guava](https://guava.dev/releases/19.0/api/docs/com/google/common/collect/ImmutableMap.Builder.html) actually). Maybe reuse that? Or is it too verbose?